core/src/storage2/collections/bitvec/iter.rs

```diff
@@ -0,0 +1,310 @@
+                    self.remaining -= 1;
+                    return front
+                }
+            }
+            match self.bits256_iter.next() {
+                None => {
+                    if let Some(back) = self.back_iter.as_mut()?.next() {
+                        self.remaining -= 1;
+                        return Some(back)
+                    }
+                    return None
+                }
+                Some(ref mut front) => {
+                    self.front_iter = Some(unsafe { extend_lifetime(front) }.iter());
+                }
+            }
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let remaining = self.remaining as usize;
+        (remaining, Some(remaining))
+    }
+
+    fn count(self) -> usize {
+        self.remaining as usize
+    }
+}
+
+impl<'a> DoubleEndedIterator for BitsIter<'a> {
+    fn next_back(&mut self) -> Option<Self::Item> {
+        loop {
+            if let Some(ref mut back_iter) = self.back_iter {
+                if let back @ Some(_) = back_iter.next_back() {
+                    self.remaining -= 1;
+                    return back
+                }
+            }
+            match self.bits256_iter.next_back() {
+                None => {
+                    if let Some(front) = self.front_iter.as_mut()?.next_back() {
+                        self.remaining -= 1;
+                        return Some(front)
+                    }
+                    return None
+                }
+                Some(ref mut back) => {
+                    self.back_iter = Some(unsafe { extend_lifetime(back) }.iter());
+                }
+            }
+        }
+    }
+}
+
+/// Iterator over the bits of a storage bit vector.
+#[derive(Debug)]
+pub struct BitsIterMut<'a> {
+    remaining: u32,
+    bits256_iter: Bits256IterMut<'a>,
+    front_iter: Option<Bits256BitsIterMut<'a>>,
+    back_iter: Option<Bits256BitsIterMut<'a>>,
+}
+
+impl<'a> BitsIterMut<'a> {
+    /// Creates a new iterator yielding the bits of the storage bit vector.
+    pub(super) fn new(bitvec: &'a mut StorageBitvec) -> Self {
+        Self {
+            remaining: bitvec.len(),
+            bits256_iter: bitvec.iter_chunks_mut(),
+            front_iter: None,
+            back_iter: None,
+        }
+    }
+}
+
+impl<'a> ExactSizeIterator for BitsIterMut<'a> {}
+
+impl<'a> Iterator for BitsIterMut<'a> {
+    type Item = BitRefMut<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            if let Some(ref mut front_iter) = self.front_iter {
+                if let front @ Some(_) = front_iter.next() {
+                    self.remaining -= 1;
+                    return front
+                }
+            }
+            match self.bits256_iter.next() {
+                None => {
+                    if let Some(back) = self.back_iter.as_mut()?.next() {
+                        self.remaining -= 1;
```

Comment:
So this is exercised in concert with `next_back()`, I think a test for this would be good. The existing ones just go forwards and backwards so wouldn't hit this or from the other direction in `DoubleEndedIterator`

Reply:
I have seen you already added a test case for this. so this can be resolved i guess.